libs/logstash-bridge/src/test/java/org/elasticsearch/logstashbridge/ThreadPoolBridgeTests.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.logstashbridge;
+
+import org.elasticsearch.common.util.concurrent.ThreadContext;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.threadpool.Scheduler;
+import org.elasticsearch.threadpool.ThreadPool;
+
+import java.util.concurrent.Executor;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static org.elasticsearch.logstashbridge.DucktypeMatchers.instancesQuackLike;
+import static org.elasticsearch.logstashbridge.DucktypeMatchers.staticallyQuacksLike;
+
+public class ThreadPoolBridgeTests extends ESTestCase {
+
+    public void testThreadPoolAPIStability() throws Exception {
```

Comment:
I don't understand the purpose of these tests. The point of this lib is that it is a bridge that will maintain all compatibility promises. If you need access to ThreadPool, have your own ThreadPool wrapper, and delegate in this bridge lib to the ES ThreadPool. By doing that, any refactorings (eg renamings/adjustments/whatever) of ThreadPool methods will require ES devs to update this code before merging, no notification to logstash devs necessary.

Reply:
Got it.

For many of these objects, we need to pass actual instances of these objects to ES-owned code, or to work with instances that are provided _by_ ES-owned code.\

I had hoped to avoid many short-lived wrappers for each object by applying some pressure to the stability of the API, but I understand now that we are not to expect that to work.

I'll convert this PR back to 'draft' and work up a wrapper pattern for each of the objects we need to work with.